lib/ohai/plugins/vmware.rb

```diff
@@ -0,0 +1,76 @@
+
+#
+# Author:: "Dan Robinson" <drobinson@getchef.com>
+# Copyright:: Copyright (c) 2014 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Provides a set of attributes for a VMware ESX virtual machine with results
+# obtained from vmware-toolbox-cmd.  VMware Tools must be installed
+# on the virtual machine.
+#
+# Modify the path to vmware-toolbox-cmd in the call to get_vm_attributes for
+# your particular operating system and configuration
+#
+# Example:
+#
+# get_vm_attributes("/usr/bin/vmware-toolbox-cmd")
+#
+
+Ohai.plugin(:Vmware) do
+  provides "vmware"
+
+  def from_cmd(cmd)
+    so = shell_out(cmd)
+    so.stdout.split($/)[0]
+  end
+
+  def get_vm_attributes(vmtools_path)
+    if !File.exist?(vmtools_path)
+      Ohai::Log.debug("#{vmtools_path} not found")
+    else
+      vmware Mash.new
+      begin
+        # vmware-toolbox-cmd stat <param> commands
+        # Iterate through each parameter supported by the "vnware-toolbox-cmd stat" command, assign value
```

Comment:
Very minor typo to fix; I see a couple of instances of 'vnware-toolbox-cmd'; do you mean vmware-toolbox-cmd? Doesn't affect functionality obviously, but is helpful for searching.